static/app/views/automations/components/dataConditionNodes.tsx

```diff
@@ -0,0 +1,143 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import NumberField from 'sentry/components/forms/fields/numberField';
+import SelectField from 'sentry/components/forms/fields/selectField';
+import {t, tct} from 'sentry/locale';
+import {
+  type DataCondition,
+  DataConditionType,
+} from 'sentry/types/workflowEngine/dataConditions';
+
+export type Node = {
+  label: string;
+  configNode?: (
+    condition: Omit<DataCondition, 'condition_group' | 'type' | 'id'>,
+    condition_id: string,
+    onUpdate: (condition: Record<string, any>) => void
```

Comment:
This callback should accept a context object so it remains extensible in case the shape of the parameters ever changes. A good rule of thumb is to use an object if there might ever reasonably be more than 2 arguments.